hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -302,6 +302,10 @@ case class HoodieFileIndex(
       PartitionRowPath(partitionRow, partitionPath)
     }
 
+    if (partitionRowPaths.isEmpty) {
+      partitionRowPaths = Seq(PartitionRowPath(InternalRow.empty, "")).toBuffer
+    }
+
```

Comment:
simple fixed the bug ,, introduce by HUDI-1591

Reply:
@xiarixiaoyao can we raise another PR to fix the bug?

Reply:
@leesf     thanks for you review.      i will raise another pr to solve this problem。